block/manager.go

```diff
@@ -984,16 +988,16 @@ func (m *Manager) getSignature(header types.Header) (*types.Signature, error) {
 	return &signature, nil
 }
 
-func (m *Manager) getTxsFromBatch() cmtypes.Txs {
+func (m *Manager) getTxsFromBatch() (cmtypes.Txs, time.Time) {
 	batch := m.bq.Next()
 	if batch == nil {
-		return make(cmtypes.Txs, 0)
+		return make(cmtypes.Txs, 0), time.Now()
```

Comment:
This implicitly requires that sequencer and rollkit use synchronized clocks, right?

Reply:
i think this is little bit tricky. there is no batch coming from the sequencer and rollkit wants to create empty headers. not sure how to tackle this. any suggestions?

Reply:
modified little bit to add empty batch to queue so that we can extract the sequencer timestamps and use it for empty batches as well. 
the only case remains is when the batch is `nil`, which can only happen if the batch retrieval is set to slow and blocktime is faster. currently this cannot happen either, as we are using blocktime as default batch retrieval time and it is not configurable presently.

Reply:
may be it is more correct to `panic` if we encounter `batch == nil` to prevent inconsistent timestamping. what we want is always have sequencer timestamp.

Reply:
Can we write a test? The test updates for this feel surface level so the fix is covering a case that we weren't testing previously I assume.

Reply:
ok. handled it slightly differently. `getTxsFromBatch` will return `ErrNoBatch` and `publishBlock` will also error out without proceeding with block creation. this should work?